sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -129,7 +129,12 @@ trait InvokeLike extends Expression with NonSQLExpression {
       // return null if one of arguments is null
       null
     } else {
-      val ret = method.invoke(obj, args: _*)
+      val ret = try {
+        method.invoke(obj, args: _*)
+      } catch {
+        // Re-throw the original exception.
+        case e: java.lang.reflect.InvocationTargetException => throw e.getCause
```

Comment:
nit: a potential NPE here:)

Reply:
do you mean `e.getCause` can be null?

Reply:
right

Reply:
I don't know how the cause can be null for `InvocationTargetException`, but let me add the null check for safety.